### PR TITLE
Multi-concentrator E2E test

### PR DESCRIPTION
**Pull request by MaggieSalak**
# PR for issue #836 

## What is being addressed

E2E test for multi-concentrator scenario using an OTAA device.

## How is this addressed

- new OTAA device defined
- new E2E test added in a new test class

## How was it tested

The test was successfully run by @danigian on a setup with 2 concentrators. Enabling it in the E2E CI would require adding another concentrator to the current setup - cc @Mandur for additional input here.

### CODE REVIEW

File: Tests/E2E/MultiConcentratorTests.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            await ArduinoDevice.setDeviceModeAsync(LoRaArduinoSerial._device_mode_t.LWOTAA);
+            await ArduinoDevice.setIdAsync(device.DevAddr, device.DeviceID, device.AppEUI);
+            await ArduinoDevice.setKeyAsync(device.NwkSKey, device.AppSKey, device.AppKey);
+
+            await ArduinoDevice.SetupLora(TestFixtureCi.Configuration);
+
+            var joinSucceeded = await ArduinoDevice.setOTAAJoinAsyncWithRetry(LoRaArduinoSerial._otaa_join_cmd_t.JOIN, 20000, 5);
+            Assert.True(joinSucceeded, "Join failed");
+
+            var logMsg = $"Duplicate message received from station with EUI";
```
Comment (spygi) [inline Tests/E2E/MultiConcentratorTests.cs:37]: Shall we make this a constant and re-use from `ConcentratorDeduplication` and here? This way if someone adjusts the log message the test won't fail unnecessarily.

Comment (MaggieSalak) [reply Tests/E2E/MultiConcentratorTests.cs]: Unfortunately we can't assert on the exact same message here as we don't know the station EUI. Removed the '$'

**Comment by Mandur** [comment]
I just tested and my rak833 is toasted. We would need to get another one. Normaly we have several of these devices. Otherwise I also have another raspberry pi hat but that would pose some logistical issues on the CI setup

**Comment by Mandur** [comment]
Question: Do you plan to enable this test on the CI later once we have the infra?

**Comment by MaggieSalak** [comment]
> Question: Do you plan to enable this test on the CI later once we have the infra?

Yes, that's the goal once the infra allows it

File: Tests/E2E/MultiConcentratorTests.cs (hunk 1 above)
Comment (MaggieSalak) [reply Tests/E2E/MultiConcentratorTests.cs]: Changed to use a constant

**Review by Mandur — APPROVED** [review]

**Comment by Mandur** [comment]
Is this work tracked or should we create additional user stories?

**Comment by MaggieSalak** [comment]
> Is this work tracked or should we create additional user stories?

It is tracked in this story #836. But maybe it makes sense to split it then since we can't enable the test in the CI right away.

**Pull request merged**